server/src/main/java/org/opensearch/action/admin/indices/scale/searchonly/ScaleIndexRequestBuilder.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.indices.scale.searchonly;
+
+import org.opensearch.action.ActionRequestBuilder;
+import org.opensearch.action.support.clustermanager.AcknowledgedResponse;
+import org.opensearch.common.annotation.PublicApi;
+import org.opensearch.transport.client.OpenSearchClient;
+
+/**
+ * A builder for constructing {@link ScaleIndexRequest} objects to perform search-only scale operations.
+ * <p>
+ * This builder simplifies the creation of requests to scale indices up or down for search-only mode.
+ * It provides methods to configure the scaling direction and follows the builder pattern to allow
+ * method chaining for constructing requests.
+ * <p>
+ * The builder is part of the public API since OpenSearch 3.0.0.
+ */
+@PublicApi(since = "3.0.0")
+public class ScaleIndexRequestBuilder extends ActionRequestBuilder<ScaleIndexRequest, AcknowledgedResponse> {
+
+    /**
+     * Constructs a new builder for scaling an index.
+     * <p>
+     * By default, the operation is a scale-up operation (from search-only to normal mode).
```

Comment:
Does it make sense to have a default? Or should we force clients to decide which operation they want?

I think we only really need the three-arg constructor, which then means we can drop the setter.

Reply:
Coming from https://github.com/opensearch-project/OpenSearch/pull/17299#discussion_r2015156859 let me remove defaults and add three-arg constructor.